# Bibliotheken-Imports & Startmeldung

In [2]:
import pandas as pd
import geopandas as gpd
import matplotlib.pyplot as plt
import os

print("Projektstart erfolgreich!")


Projektstart erfolgreich!


# Pfade definieren & Datei einlesen

## Automatisches Finden einer Beispieldatei
In diesem Abschnitt wird eine geeignete Beispieldatei aus den BASt‑Rohdaten automatisch ausgewählt und eingelesen.
Das erleichtert die Arbeit, weil man nicht jedes Mal einen Dateinamen manuell angeben muss.

In [8]:
# Pfad zu den 2025-Daten:
# Hier wird festgelegt, in welchem Ordner sich die Rohdaten befinden. 
data_dir = "C:/Users/Marijke/iCloudDrive/Dokumente/AKAD MS Data Science/GitHub/verlagerungsmodell_gueterverkehr/data/raw/zaehlstellen/2025/Roh_2025_A_S/"

# Automatisch eine Beispiel-CSV finden (erste Monatsdatei)
# Die Schleife sucht die erste Datei, die keine Metadaten-Datei ist (daher der Ausschluss
# ["meta" not in f.lower()]. Das Ergebnis wird in example_file gespeichert
example_file = None
for f in os.listdir(data_dir):     # Auflistung aller Dateien im Ordner
    if f.lower().endswith(".csv") and "meta" not in f.lower():
        example_file = f
        break

print("Gefundene Beispieldatei:", example_file) 

# Erzeugung des vollständigen Dateipfades
# [os.path.join] sorgt dafür, dass der Pfad unabhängig vom Betriebssystem korrekt zusammengesetzt wird
path = os.path.join(data_dir, example_file)

Gefundene Beispieldatei: Roh_2025_01_A_S.csv


## Datei einlesen

In [10]:
df = pd.read_csv(path, sep=";", encoding="latin1")
print("Rohdaten geladen:", df.shape)

df.head()

Rohdaten geladen: (1067640, 57)


,TKNR,Zst,Land,Strklas,Strnum,Datum,Wotag,Fahrtzw,Stunde,KFZ_R1,...,Bus_R2,K_Bus_R2,LoA_R2,K_LoA_R2,Lzg_R2,K_Lzg_R2,Sat_R2,K_Sat_R2,Son_R2,K_Son_R2
0,2123,1154,1,A,23,250101,3,,1,0,...,0,a,0,a,0,a,0,a,0,a
1,2123,1154,1,A,23,250101,3,,2,0,...,0,a,0,a,0,a,0,a,0,a
2,2123,1154,1,A,23,250101,3,,3,0,...,0,a,0,a,0,a,0,a,0,a
3,2123,1154,1,A,23,250101,3,,4,0,...,0,a,0,a,0,a,0,a,0,a
4,2123,1154,1,A,23,250101,3,,5,0,...,0,a,0,a,0,a,0,a,0,a


# Data Dictionary – BASt Richtungsaggregierte Rohdaten 2025

Dieses Data Dictionary beschreibt die wichtigsten Spalten der BASt-Rohdaten (Autobahnen, richtungsaggregiert). Es dient als Referenz für die Datenexploration, Modellierung und spätere Dashboard-Entwicklung

## 🧭 Identifikationsspalten

- **TKNR** – Technische Knotennummer (interne BASt-ID der Messstelle)  
- **Zst** – Zählstellen-ID (verbindet Rohdaten mit Metadaten)  
- **Land** – Bundesland (numerischer Code, z. B. 1 = BW, 5 = NRW)  
- **Strklas** – Straßenklasse (A = Autobahn, B = Bundesstraße)  
- **Strnum** – Straßennummer (z. B. 23 = A23)  

---

## 🕒 Zeitliche Informationen

- **Datum** – Datum im Format `YYMMDD` (z. B. 250101 = 01.01.2025)  
- **Wotag** – Wochentag (1 = Montag … 7 = Sonntag)  
- **Stunde** – Stunde des Tages (1–24)  
- **Fahrtzw** – Fahrtrichtung (1 oder 2; manchmal leer)  

---

## 🚗 Verkehrsmengen Richtung 1 (R1)

- **KFZ_R1** – Gesamtzahl aller Kraftfahrzeuge Richtung 1  
- **Bus_R1** – Busse Richtung 1  
- **LoA_R1** – Lkw über 3,5 t Richtung 1  
- **Lzg_R1** – Lastzüge Richtung 1  
- **Sat_R1** – Sattelzüge Richtung 1  
- **Son_R1** – Sonstige schwere Fahrzeuge Richtung 1  

---

## 🚚 Verkehrsmengen Richtung 2 (R2)

- **KFZ_R2** – Gesamtzahl aller Kraftfahrzeuge Richtung 2  
- **Bus_R2** – Busse Richtung 2  
- **LoA_R2** – Lkw über 3,5 t Richtung 2  
- **Lzg_R2** – Lastzüge Richtung 2  
- **Sat_R2** – Sattelzüge Richtung 2  
- **Son_R2** – Sonstige schwere Fahrzeuge Richtung 2  

---

## 🏷️ Qualitätskennzeichen (K_…)

Zu jeder Verkehrsspalte existiert eine Qualität

# Data Dictionary – BASt Metadaten (Zählstelleninformationen)

Die Metadaten enthalten alle statischen Informationen zu den automatischen Dauerzählstellen.  
Sie verbinden die Rohdaten (Verkehrsmengen) mit geografischen und infrastrukturellen Eigenschaften der Messstellen.

---

## 🧭 Identifikations- und Verwaltungsinformationen

- **Zst** – Zählstellen-ID (eindeutige Kennung der Messstelle; verbindet Rohdaten und Metadaten)  
- **TKNR** – Technische Knotennummer (interne BASt-ID)  
- **Strklas** – Straßenklasse (A = Autobahn, B = Bundesstraße)  
- **Strnum** – Straßennummer (z. B. 23 = A23)  
- **Strname** – Straßenname oder Bezeichnung (falls vorhanden)  
- **Land** – Bundesland (numerischer Code, z. B. 1 = BW, 5 = NRW)  

---

## 🗺️ Geografische Informationen

- **Latitude** – geografische Breite der Zählstelle (WGS84)  
- **Longitude** – geografische Länge der Zählstelle (WGS84)  
- **Abschnitt** – Abschnittsnummer der Straße (falls vorhanden)  
- **Station** – Stationsnummer innerhalb des Abschnitts (falls vorhanden)  

Diese Spalten ermöglichen die Darstellung der Zählstellen auf Karten (z. B. GeoPandas, Folium).

---

## 🚦 Fahrtrichtungs- und Infrastrukturinformationen

- **Fahrtrichtung** – beschreibt die Richtung der Messung (z. B. „Richtung Hamburg“)  
- **Richtung** – numerische Richtungskennung (1 oder 2)  
- **Fahrstreifen** – Anzahl der Fahrstreifen an der Messstelle (falls vorhanden)  
- **Zst_Typ** – Typ der Messstelle (z. B. Induktionsschleife, Radar)  

---

## 🏗️ Weitere Merkmale

- **Inbetriebnahme** – Datum der Inbetriebnahme der Messstelle  
- **Stilllegung** – Datum der Stilllegung (falls nicht mehr aktiv)  
- **Kommentar** – zusätzliche Hinweise der BASt (z. B. Umbauten, Besonderheiten)  

---

## 🔗 Verbindung zu den Rohdaten

Die Metadaten werden über die Spalte **Zst** mit den Rohdaten verknüpft.  
Beispiel:

```python
df_merged = df.merge(meta, on="Zst", how="left")
